code/modules/food_and_drinks/kitchen_machinery/cooker.dm

```diff
@@ -41,15 +41,24 @@
 /obj/machinery/cooker/proc/checkValid(obj/item/check, mob/user)
 	if(on)
 		to_chat(user, "<span class='notice'>[src] is still active!</span>")
-		return 0
-	if(istype(check, /obj/item/reagent_containers/food/snacks))
-		return 1
+		return FALSE
 	if(istype(check, /obj/item/grab))
 		return special_attack(check, user)
 	if(has_specials && checkSpecials(check))
 		return TRUE
-	to_chat(user, "<span class ='notice'>You can only process food!</span>")
-	return 0
+	if(istype(check, /obj/item/reagent_containers/food/snacks) || emagged)
+		if(istype(check, /obj/item/disk/nuclear)) //(1984 voice) you will not deep fry the NAD
+			to_chat(user, "<span class='notice'>The disk is more useful raw than [thiscooktype].</span>")
+			return FALSE
+		var/obj/item/disk/nuclear/datdisk = locate() in check
+		if(datdisk)
+			to_chat(user, "<span class='notice'>You get the feeling that something very important is inside this. Something that shouldn't be [thiscooktype].</span>")
+			return FALSE
+		if(check.flags & (ABSTRACT || DROPDEL || NODROP)) //you will not deep fry the armblade
```

Comment:
You'll want to use bitwise operators when comparing flags.

```suggestion
		if(check.flags & (ABSTRACT | DROPDEL | NODROP)) //you will not deep fry the armblade
```

Reply:
Thanks for the heads up, fixed!